owners/src/local_repo.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const path = require('path');
+const fs = require('fs').promises;
+const child_process = require('child_process');
+const util = require('util');
+const exec = util.promisify(child_process.exec);
+
+/**
+ * Execute raw shell commands.
+ *
+ * @private
+ * @param {!!string[]} commands list of commands to execute.
+ * @return {string[]} command output
+ */
+async function runCommands(...commands) {
+  return await new Promise(async (resolve, reject) => {
+    const {stdout, stderr} = await exec(commands.join(' && '));
+    if (stderr) {
+      reject(stderr);
+    } else {
+      resolve(stdout);
+    }
+  });
+}
+
+
+/**
+ * Interface for reading from a checked out repository using relative paths.
+ */
+class LocalRepository {
+  /**
+   * Constructor.
+   *
+   * @param {!string} pathToRepoDir absolute path to the repository root
+   *     directory.
+   * @param {!string=} remote git remote to clone (default: 'origin').
+   */
+  constructor(pathToRepoDir, remote) {
+    this.rootDir = pathToRepoDir;
+    this.remote = remote || 'origin';
+  }
+
+  /**
+   * Runs commands in the repository's root directory.
+   *
+   * @param {!string[]} commands list of commands to execute.
+   * @return {string[]} command output
+   */
+  async runCommands(...commands) {
+    return await runCommands(`cd ${this.rootDir}`, ...commands);
+  }
+
+  /**
+   * Check out a branch locally.
+   *
+   * @param {!string=} branch git branch to checkout (default: 'master').
+   */
+  async checkout(branch) {
+    branch = branch || 'master';
+    await this.runCommands(
+        `git fetch ${this.remote} ${branch}`,
+        `git checkout -B ${branch} ${this.remote}/${branch}`);
+  }
+
+  /**
+   * Get an absolute path for a relative repo path.
+   *
+   * @param {!string} relativePath file or directory path relative to the root.
+   * @return {string} absolute path to the file in the checked out repo.
+   */
+  getAbsolutePath(relativePath) {
+    return path.resolve(this.rootDir, relativePath);
+  }
+
+  /**
+   * Read the contents of a file from the checked out repo.
+   *
+   * @param {!string} relativePath file to read.
+   * @return {string} file contents.
+   */
+  async readFile(relativePath) {
+    const filePath = this.getAbsolutePath(relativePath);
+    return await fs.readFile(filePath, {encoding: 'utf8'});
```

Comment:
You can use `fs.readFileSync` and remove the async keyword here (unless there's a good use-case for it to be async of course) - if you change that, change the callers to not `await` on this (`await`ing on a non-async function is harmless though, so nbd if you miss one)

Reply:
I considered that, but the other files in the app were using `require('fs').promises` so I left it. I'll come back to this and sync-ify.

Reply:
Done